### ISSUE TITLE
Rework of how eventghost handles debugging

### ISSUE THREAD

**Issue by kdschlosser** [EventGhost/EventGhost#154]
I have a question.

I have been working on rewriting the whole logging bits of EG.

This is what I have done thus far.

EG had in place multiple storage containers for logging. I have gotten rid of all but one. and also instituted a means so the user can set how many lines to store. which is a sweet deal :-D no more scrolling back to find the log has ended after 2000 entries. 

On the debugging end of things. 
I added a storage for this as well as the same option to set the depth of the storage. I did this for one sole purpose. example. a traceback is thrown. oooops crap i didn't have debugging enabled to really see what is going on, with how i have set it up you can simply enable debugging to gain access to the past debugging data. and if you do not want this feature activated you can simply squash the queue size down to 0 and at a 0 it will shut off this feature entirely.

I also added a nice debugging console if you will. it parses the file as well as reads any new data
and you now have the ability to empty the log while EG is running. YAY!!! and it provides a nice little progress indicator when reading the file just in case your file is rather large. and you can cancel the loading and it will display what has loaded thus far. I plan on making a resume for the loading as well. haven't gotten that far yet.

a search function is on list of things to add.

what I would like to know is are there any features or options that anyone would like to see added?  

![debugging](https://cloud.githubusercontent.com/assets/10932297/21598135/3fc437ce-d114-11e6-9438-1b49b6eddaed.png)

**Comment by topic2k** [comment]
Sounds really good. Now i don't have any ideas for something more, but maybe that will change if i have something to play with ;)

Oh, just one question: The debugging console, is it the debug log in the screenshot or not shown in the screenshot?

**Comment by kdschlosser** [comment]
It's shown. the bottom pane in the UI and that extra open window on the right. the window provides the same details as the log pane but offers a means to put 2 items next to each other that could be at opposite ends of the log. if you needed to compare information.

**Comment by kdschlosser** [comment]
it parses the log and breaks it rown into date/time and makes a TreeListCtrl with the date and times as tree items. and lists the actual debug entries under the times. but I have it read the line and look for specific things to add a "tag" to it weather it's just a call with no return. or there is a return. informational or an error. I already have set up the means to actually put the exception name if it's a traceback. and it will pull the doc string on the exception so the user can read what the exception actually is.  The really nice thing is to not have to go to the directory and close EG to delete the log. I coded in the means to close the open file and then reopen it with a "w" if it gets cleared so it will be over written.. it will only do this if you clear the log. so upon restart of EG it will go back to an "a". I am trying to find the best means with the whole log as far as the cache's I made a single container but because of the use of callafter and deleting any otems over a specified size the multiple threads is giving me a hard hour. of i use locks. the performance drops. and with the destruction of the frame on minimize it's seeming like the only viable option is to have 2 cache's. maybe the best way is to not add to the first one at all unless the dialog is closed. st the current moment it always uses the first and second.
